### PR TITLE
[UPDATE][RAYL] Validation added

### PR DESCRIPTION
**Pull request by rayleeeigh**
**What does this PR do?**

**Where should the reviewer start?**

**Related ticket link/s:**

**Before and After UI Screenshot (optional):**

### CODE REVIEW

File: client/src/pages/ResetPassword.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import { Box, Input, Container, Button, Center, Heading, FormControl, FormLabel, Text } from "@chakra-ui/react";
+import Axios from 'axios';
+import { useForm } from "react-hook-form";
+
+const ResetPassword = () => {
+    const [show1, setShow1] = useState(false);
+    const handleClick1 = () => setShow1(!show1)
+
+    const [show2, setShow2] = useState(false);
+    const handleClick2 = () => setShow2(!show2)
```
Comment (lourencelinao) [inline client/src/pages/ResetPassword.js:11]: make this variable descriptive

File: client/src/pages/ResetPassword.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
+import { Box, Input, Container, Button, Center, Heading, FormControl, FormLabel, Text } from "@chakra-ui/react";
+import Axios from 'axios';
+import { useForm } from "react-hook-form";
+
+const ResetPassword = () => {
+    const [show1, setShow1] = useState(false);
+    const handleClick1 = () => setShow1(!show1)
```
Comment (lourencelinao) [inline client/src/pages/ResetPassword.js:8]: make this variable descriptive

File: client/src/pages/ResetPassword.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    const [show2, setShow2] = useState(false);
+    const handleClick2 = () => setShow2(!show2)
+  
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [confirmpassword, setConfirmPassword] = useState("");
+
+    const { register, handleSubmit, formState: { errors } } = useForm();
+    const onSubmit = (data) => alert("Reset Successful");
```
Comment (lourencelinao) [inline client/src/pages/ResetPassword.js:18]: use Toast

File: client/src/pages/ResetPassword.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            <Heading size="md">Reset Password</Heading>
+          </Container>
+          <form onSubmit={handleSubmit(onSubmit)}>
+          <Container mt="5%">
+              <Container>
+                <FormControl id="email" isRequired>
+                  <FormLabel>Email Address</FormLabel>
+                  <Input name="email" {...register("email", { required: true, pattern: {value: /^\S+@\S+$/i, message: "Invalid Email"} })}  placeholder="Enter email..." variant="filled" onChange={(event)=>{setEmail(event.target.value);}}/>
+                </FormControl>
+                {errors.email &&  (
```
Comment (lourencelinao) [inline client/src/pages/ResetPassword.js:43]: avoid logic inside the jsx, assign this to a variable

File: client/src/pages/ResetPassword.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                    <Text color="red" fontSize="xs">{errors.email.message}</Text>
+                )}
+              </Container>
+              <Container>
+                <FormControl id="password" isRequired>
+                  <FormLabel>Password</FormLabel>
+                  <Input type={show1 ? "text" : "password"} name="password" aria-invalid={errors.password ? "true" : "false"} {...register("password", { required: "Field must not be empty", minLength: 8 })} placeholder="Enter password" variant="filled" onChange={(event)=>{setPassword(event.target.value);}}/>
+                  <Button position="absolute" ml="-10vh" mt="1vh" h="1.75rem" size="sm" onMouseDown={handleClick1} onMouseUp={handleClick1}>
+                    {show1 ? "Hide" : "Show"}
```
Comment (lourencelinao) [inline client/src/pages/ResetPassword.js:52]: avoid logic inside the jsx, assign this to a variable

File: client/src/pages/ResetPassword.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                  </Button>
+                </FormControl>
+                {errors.password &&  (
```
Comment (lourencelinao) [inline client/src/pages/ResetPassword.js:55]: avoid logic inside the jsx, assign this to a variable

File: client/src/pages/ResetPassword.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+                    <Text color="red" fontSize="xs">{errors.password.message}</Text>
+                )}
+                {errors.password && errors.password.type === "minLength" && (
+                    <Text color="red" fontSize="xs">Password must have atleast 8 characters</Text>
+                )}
```
Comment (lourencelinao) [inline client/src/pages/ResetPassword.js:60]: avoid logic inside the jsx, assign this to a variable

File: client/src/pages/ResetPassword.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+              </Container>
+              <Container>
+                <FormControl id="password" isRequired>
+                  <FormLabel>Confirm Password</FormLabel>
+                  <Input name="confirmpassword" aria-invalid={errors.confirmpassword ? "true" : "false"} {...register("confirmpassword", { required: "Field must not be empty", minLength: 8, validate: value => value === password.current || "Passwords do not match" })} type={show2 ? "text" : "password"} placeholder="Confirm password" variant="filled" onChange={(event)=>{setConfirmPassword(event.target.value);}}/>
+                  <Button position="absolute" ml="-10vh" mt="1vh" h="1.75rem" size="sm" onMouseDown={handleClick2} onMouseUp={handleClick2}>
+                    {show2 ? "Hide" : "Show"}
```
Comment (lourencelinao) [inline client/src/pages/ResetPassword.js:67]: avoid logic inside the jsx, assign this to a variable

File: client/src/pages/ResetPassword.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+                  </Button>
+                </FormControl>
+                {errors.confirmpassword &&  (
+                    <Text color="red" fontSize="xs">{errors.confirmpassword.message}</Text>
+                )}
```
Comment (lourencelinao) [inline client/src/pages/ResetPassword.js:72]: avoid logic inside the jsx, assign this to a variable

File: client/src/pages/ResetPassword.js [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+                {errors.confirmpassword && errors.confirmpassword.type === "minLength" && (
+                    <Text color="red" fontSize="xs">Password must have atleast 8 characters</Text>
+                )}
```
Comment (lourencelinao) [inline client/src/pages/ResetPassword.js:75]: avoid logic inside the jsx, assign this to a variable

File: server/routes/accountRoutes.js [hunk 11]
```diff
@@ -4,5 +4,6 @@ const accountsController = require("../controllers/accountsController");
 
 router.post("/create", accountsController.createAccount);
 router.get("/accounts", accountsController.showAccounts);
+router.post("/resetpassword",accountsController.resetPassword);
```
Comment (lourencelinao) [inline server/routes/accountRoutes.js:7]: add space after comma

**Review by lourencelinao — CHANGES_REQUESTED** [review]

File: client/src/pages/ResetPassword.js (hunk 1 above)
Comment (rayleeeigh) [reply client/src/pages/ResetPassword.js:11]: resolved

File: client/src/pages/ResetPassword.js (hunk 2 above)
Comment (rayleeeigh) [reply client/src/pages/ResetPassword.js:8]: resolved

File: client/src/pages/ResetPassword.js (hunk 3 above)
Comment (rayleeeigh) [reply client/src/pages/ResetPassword.js:18]: resolved

**Review by arnan123 — APPROVED** [review]

**Pull request merged**